xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/stylesheets.vm

```diff
@@ -21,9 +21,9 @@
 
 #macro(concatenateQueryParameter $originalUrl $queryString)
   #if ($originalUrl.contains("?"))
-    $originalUrl&$queryString
+$originalUrl&#38;$queryString
   #else
-    $originalUrl?$queryString
+$originalUrl?$queryString
```

Comment:
Wrong indentation?

Reply:
This was an attempt to prevent additional whitespace in the output. Though in fact this is not enough, I should have removed that change again as it doesn't actually fix the issue. What would actually be needed is

```velocity
#if($originalUrl.contains("?"))$originalUrl&#38;$queryString#else$originalUrl?$queryString#end
```

With proper indentation this code leads to

```html
<link href="      /xwiki/bin/skin/skins/flamingo/style.min.css?cache-version=1643188958000&#38;skin=XWiki.DefaultSkin&#38;colorTheme=xwiki%3AFlamingoThemes.Iceberg&#38;colorThemeVersion=1.1
  " rel="stylesheet" type="text/css" media="all" />
<link href="      /xwiki/bin/skin/skins/flamingo/print.min.css?cache-version=1643188958000&#38;skin=XWiki.DefaultSkin&#38;colorTheme=xwiki%3AFlamingoThemes.Iceberg&#38;colorThemeVersion=1.1
  " rel="stylesheet" type="text/css" media="print" />
```

I don't think this is invalid but it definitely doesn't look nice. Note how this macro also introduces a linebreak in the attribute. Even the indentation of the `#if` creates whitespace.

Reply:
I'm not sure it's applicable in this case but it's possible to prevent line-breaks by adding a comment at the end of a line.
For instance:
```velocity
$originalUrl&#38;$queryString##
```

Reply:
It is working, so we can prevent the line break. All indentation causes spaces to appear, so if we have properly indented code we will have whitespace in the output.

I've just checked, it is definitely valid [with whitespace](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#valid-non-empty-url-potentially-surrounded-by-spaces) so it is on us if we want to have cleaner code or cleaner output.

Reply:
I'm leaning toward cleaner output. A comment explaining why we don't follow the usual formatting could be interesting though.